code/modules/clothing/_clothing.dm

```diff
@@ -179,3 +179,31 @@
 		for(var/key in damages)
 			to_chat(user, "<li><b>[capitalize(damages[key])]</b> damage to the <b>[key]</b> armor.")
 		return TOPIC_HANDLED
+
+/obj/item/clothing/get_pressure_weakness(pressure,zone)
+	. = 1
+	if(pressure > ONE_ATMOSPHERE)
+		var/max_pressure = max_pressure_protection
+		if(accessories.len)
```

Comment:
Couldn't you calc and cache this when an accessory is added or removed?

Reply:
lemme give it a try